### ISSUE TITLE
Feature Request: Preserve Source IP

### ISSUE THREAD

**Issue by passarela** [microsoft/azure-container-apps#726]
**Is your feature request related to a problem? Please describe**.  

When receiving requests in my container, I cannot know which are the IP's of origin.

**Describe the solution you'd like**.    

Through the Ingress settings, Enable to keep the source IP.

**Describe alternatives you've considered**.   
I installed TCPDUMP to collect incoming requests in my container, however as it is behind an Azure LoadBalancer or source IP it is not preserved.

I expect similar behavior when setting ````externalTrafficPolicy: Local```` on a Kubernetes LoadBalancer.

**Additional context**.  
![image](https://user-images.githubusercontent.com/72925771/233464871-771325dc-aa40-452b-8ea9-9fd50cdb3386.png)

**Comment by ahmelsayed** [comment]
For http apps, you can use proxy protocol (i.e XFF headers) for the source IP as observed by the proxy. See for [nginx](https://docs.nginx.com/nginx/admin-guide/load-balancer/using-proxy-protocol/) or [haproxy](https://www.haproxy.com/blog/use-the-proxy-protocol-to-preserve-a-clients-ip-address/) or for [asp.net](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-7.0)

**Comment by passarela** [comment]
> Para aplicativos http, você pode usar o protocolo proxy (ou seja, cabeçalhos XFF) para o IP de origem conforme observado pelo proxy. Consulte [nginx](https://docs.nginx.com/nginx/admin-guide/load-balancer/using-proxy-protocol/) ou [haproxy](https://www.haproxy.com/blog/use-the-proxy-protocol-to-preserve-a-clients-ip-address/) ou [asp.net](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-7.0)

Yes, I'm aware of the XFF header, but when followed with Kubernetes, the Loadbalancer needs to be set to ``externalTrafficPolicy: Local``.

https://techcommunity.microsoft.com/t5/fasttrack-for-azure/how-client-source-ip-preservation-works-for-loadbalancer/ba-p/3033722

**Comment by ahmelsayed** [comment]
That's for the loadbalancer target, which in ACA case is a proxy that sets that information in XFF headers.

**Comment by alanta** [comment]
XFF headers work for HTTP/HTTPS but not for other TCP services. For example, I'm working on a app that receives messages over SMTP. This requires the application to be able to get the client IP to perform further checks before accepting incoming messages.

**Comment by codylittle** [comment]
Since envoy is used, the support for Proxy Protocol (envoy.transport_sockets.upstream_proxy_protocol) would be great. Currently the lack of source IP preservation on TCP workloads is quite the blocker for us.

**Comment by dwgreen1** [comment]
I am also interested in this feature, fwiw.

**Comment by MichaMican** [comment]
Same issue here - makes Containerapps unusable for us in the current state - @ahmelsayed any update on this?

**Comment by skemp42** [comment]
Same issue. Any update on this?

**Comment by bdassow-gk** [comment]
Another bump for this feature. Would be really appreciated.
